### PR TITLE
Synonym & Prefix Fixes

### PR DESCRIPTION
**Pull request by alongreyber**
Fixed bug with synonyms, and switched prefixes to come from BMT

### CODE REVIEW

File: strider/util.py [hunk 1]
```diff
@@ -15,19 +15,42 @@ def snake_to_camel(s):
...
+            value = f(lookup)
+            if value is None:
+                raise KeyError
+            else:
+                return value
+
+        def __contains__(self, lookup):
+            return f(lookup) is not None
+
+    map_instance = Mapping()
```
Comment (patrickkwang) [inline strider/util.py:34]: I'd just `return Mapping()`.

File: tests/helpers/utils.py [hunk 2]
```diff
@@ -140,12 +140,18 @@ def normalizer_data_from_string(s):
...
-    category_mappings = defaultdict(list)
-    synset_mappings = defaultdict(list)
+    category_mappings = {}
+    synset_mappings = {}
     for line in s.splitlines():
         tokens = line.split(" ")
 
         curie = tokens[0]
+        if curie not in category_mappings:
+            category_mappings[curie] = []
```
Comment (patrickkwang) [inline tests/helpers/utils.py:150]: This is exactly what `defaultdict(list)` does, right? Why the change?

File: tests/helpers/utils.py [hunk 3]
```diff
@@ -140,12 +140,18 @@ def normalizer_data_from_string(s):
...
+        if curie not in synset_mappings:
+            # Curies are always self-synonyms
+            synset_mappings[curie] = [curie]
```
Comment (patrickkwang) [inline tests/helpers/utils.py:153]: Is line 160 now unreachable? Why do this instead of `defaultdict`? What happens if `curie` _is_ in `synset_mappings`? Is that possible, and would it then not be its own synonym?

File: strider/util.py [hunk 4]
```diff
@@ -87,6 +110,17 @@ def predicate_inverse(self, predicate):
             return self.old_case_to_new_case(predicate_inverse_old_format)
         return None
 
+    def entity_prefixes(self, entity):
+        """ Get prefixes for a given entity """
+        old_format = self.new_case_to_old_case(entity)
+        element = self.bmt.get_element(old_format)
```
Comment (patrickkwang) [inline strider/util.py:116]: How fast is this? We could memoize this function if it's helpful. Then we'd have a map pretending to be a function pretending to be a map!

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: tests/helpers/utils.py (hunk 3 above)
Comment (alongreyber) [reply tests/helpers/utils.py:153]: Our code relies on the normalizer returning the given curie in the list of equivalent_identifiers. I confirmed that this is the case in the real node normalizer:

Screen Shot 2021-04-07 at 1 33 33 PM

This was the bug that caused test_issue_102 to fail. If the curie is not listed in equivalent_identifiers we never do any synonymizing. You're correct that we can use a defaultdict to streamline this and that line 160 is unreachable.

**Review by patrickkwang — APPROVED** [review]
